src/core/geometry/qgstriangle.cpp

```diff
@@ -107,6 +107,13 @@ bool QgsTriangle::operator!=( const QgsTriangle &other ) const
   return !operator==( other );
 }
 
+QgsTriangle *QgsTriangle::newSameGeometry() const
+{
+  auto result = new QgsTriangle();
+  result->mWkbType = mWkbType;
```

Comment:
Hm - i think we have a bug here. When the triangle class was introduced the corresponding types weren't added to QgsWkbTypes!

Reply:
It copies the type of the original object (whichever it is). If it is Triangle, fine. If it is Polygon, no problem. ^^

Reply:
Sorry, I should have been clearer. It's not a bug in your code, it's an existing bug and I was just "thinking aloud".